app/src/main/java/de/danoeh/antennapod/fragment/CoverFragment.java

```diff
@@ -100,8 +139,83 @@ private void displayMediaInfo(@NonNull Playable media) {
                 + "\u00A0"
                 + StringUtils.replace(StringUtils.stripToEmpty(pubDateStr), " ", "\u00A0"));
         txtvEpisodeTitle.setText(media.getEpisodeTitle());
-        displayedChapterIndex = -2; // Force refresh
-        displayCoverImage(media.getPosition());
+        displayedChapterIndex = -1;
+        refreshChapterData(ChapterUtils.getCurrentChapterIndex(media, media.getPosition()));
+        toggleDescriptionButton();
+    }
+
+    private void toggleDescriptionButton() {
```

Comment:
How about `updateDescriptionButtonVisibility`? Toggle sounds like it always switches the state.